x-pack/plugins/monitoring/init.js

```diff
@@ -35,8 +35,9 @@ export const init = (monitoringPlugin, server) => {
     }
 
     if (config.get('xpack.monitoring.kibana.collection.enabled')) {
-      const collectorSet = createCollectorSet(monitoringPlugin.kbnServer, server); // instantiate an object for collecting/sending metrics and usage stats
-      server.expose('collectorSet', collectorSet); // expose the collector set object on the server. other plugins will call statsCollectors.register(collector) to define their own collection
+      const { collectorSet, bulkUploader } = initKibanaMonitoring(monitoringPlugin.kbnServer, server); // instantiate objects for collecting stats and uploading stats
```

Comment:
Since `initKibanaMonitoring` (as this PR writes it) can return `undefined`, doesn't this fail hard in that scenario?